modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
Given the original implementation, returning doing a toString when there is no matching type is probably safer.

An UnsupportedException is likely more correct, but runs the risk of breaking backwards compatibility.

I'd say return a supported number where applicable, and otherwise do a toString().